db/db_log_iter_test.cc

```diff
@@ -191,15 +191,12 @@ TEST_F(DBTestXactLogIterator, TransactionLogIteratorCorruptedLog) {
     ASSERT_OK(dbfull()->Flush(FlushOptions()));
     ASSERT_OK(dbfull()->FlushWAL(false));
     // Corrupt this log to create a gap
-    ROCKSDB_NAMESPACE::VectorLogPtr wal_files;
+    VectorLogPtr wal_files;
     ASSERT_OK(dbfull()->GetSortedWalFiles(wal_files));
+    ASSERT_FALSE(wal_files.empty());
     const auto logfile_path = dbname_ + "/" + wal_files.front()->PathName();
-    if (mem_env_) {
-      mem_env_->Truncate(logfile_path, wal_files.front()->SizeFileBytes() / 2);
-    } else {
-      ASSERT_EQ(0, truncate(logfile_path.c_str(),
-                   wal_files.front()->SizeFileBytes() / 2));
-    }
+    test::TruncateFile(env_, logfile_path,
```

Comment:
Should we use `mem_env_` when necessary?
I guess this is currently ok because `SpecialEnv` forwards calls to the underlying mem env. In the future, if `SpecialEnv` impl changes, will this break?

Reply:
I believe the only reason `mem_env_` had a special branch there because `truncate` only works with actual on-disk files. I think `test::TruncateFile` would cover both cases.